### PR TITLE
Add OS Compatibility

### PR DESCRIPTION
**Pull request by andrewasche**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -95,22 +122,22 @@ def get_host_ips():
     return ip_list
 
 
-def log_startup_message(port_num, is_https_enabled, host):
+def log_startup_message(port_num, is_https_enabled, hostname):
```
Comment (shubham1172) [inline main.py:125]: Why this change? Can we revert this back (to keep the PR to minimum changes)?

File: main.py [hunk 2]
```diff
@@ -53,6 +56,30 @@ async def websocket_endpoint(websocket: WebSocket):
             break
 
 
+@app.get("/metadata")
+def return_features(user_agent: Union[str, None] = Header(default=None)):
```
Comment (shubham1172) [inline main.py:60]:
```suggestion
def metadata():
```

File: main.py [hunk 3]
```diff
@@ -53,6 +56,30 @@ async def websocket_endpoint(websocket: WebSocket):
...
+    """Handle metadata for OS compatilibity"""
+    if "windows" in user_agent.lower():
```
Comment (shubham1172) [inline main.py:62]: Instead of user agent operating system, i.e. the browser's operating system, we need to decide features based on the server's operating system. For example, `hscroll` is not supported when remote-play server is running on Windows. We can use python's `sys.platform` to check this.

**Review by shubham1172 — CHANGES_REQUESTED** [review]
Hello @andrewasche, thanks for your PR! Left a few notes. We can separate out the UI part, but if you want to take a stab you can also do that in the same PR. The client needs to query `GET /metadata`, and conditionally show things.

File: main.py (hunk 1 above)
Comment (andrewasche) [reply main.py:125]: The linter was suggesting something, Sorry didn't know

File: main.py (hunk 3 above)
Comment (andrewasche) [reply main.py]: sounds good,

File: main.py (hunk 1 above)
Comment (andrewasche) [reply main.py]: reverting back sounds good

**Comment by TigreModerata** [comment]
Hi, I'm looking at this and I'm noticing that I can't control the volume on linux (manjaro) - I'm trying it on youtube because I don't have any of the netflix/prime/etc subscriptions, so maybe it's just youtube? Of course I can control the volume via touchpad. Anyone else notice this?

**Comment by shubham1172** [comment]
> Hi, I'm looking at this and I'm noticing that I can't control the volume on linux (manjaro) - I'm trying it on youtube because I don't have any of the netflix/prime/etc subscriptions, so maybe it's just youtube? Of course I can control the volume via touchpad. Anyone else notice this?

It might be related to https://github.com/asweigart/pyautogui/issues/443

**Comment by shubham1172** [comment]
Closing this PR in favor of #47, which contains the client-side changes as well. Thanks @andrewasche again for the PR! 🙂
